### PR TITLE
MODINVOSTO-48 - Update acq-models and add system data

### PR DESCRIPTION
**Pull request by craigmcnally**
## Purpose
In order to support batch vouchers, several new schemas were added to acq-models.  This PR updates the acq-submodule, pulling in those schema changes.  

Also part of this PR is the introduction of "system" data, which should always be present in the system.  See [MODINVOSTO-48](https://issues.folio.org/browse/MODINVOSTO-48) 

The final part of this PR is data-migration scripts.  With the addition of a new required field in both invoice and voucher schemas, scripts are needed to "fix" existing records which won't have those fields. 

See the [wiki](https://wiki.folio.org/display/FOLIJET/Batch+Voucher+Exports) for additional information.

## Approach
* Update the acq-models submodule
* Create system data (the "FOLIO" batch-group)
* Implement data migration scripts that set "batchGroupId" in voucher/invoice to the UUID of the FOLIO batch-group.
* In order to test that the default value is applied for batchGroupId, I removed this field from the sample data voucher and one of the sample invoices.  

## TODOS and Open Questions
- [ ] coordinate with https://github.com/folio-org/acq-models/pull/218
- [ ] A follow-on PR will be required to actually load the system data.  It's not possible to do so until [MODINVOSTO-50](https://issues.folio.org/browse/MODINVOSTO-50) is done.  This work may be split out into a separate story depending on timing and availability of related PRs.  See https://github.com/folio-org/mod-invoice-storage/pull/54

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [X] Code coverage on new code is 80% or greater
  - [X] Duplications on new code is 3% or less
  - [X] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [X] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by craigmcnally** [comment]
$ python3 ~/git/folio-tools/lint-raml/lint_raml_cop.py -ldebug -j
DEBUG: urllib3.connectionpool: Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG: urllib3.connectionpool: https://raw.githubusercontent.com:443 "GET /folio-org/folio-org.github.io/master/_data/api.yml HTTP/1.1" 200 2465
INFO: lint-raml: Investigating and determining configuration: mod-invoice-storage/ramls
DEBUG: lint-raml: ramls_dir=./ramls
DEBUG: lint-raml: Looking for JSON schema files: ./ramls/acq-models/mod-invoice-storage/schemas
DEBUG: lint-raml: found_schema_files: ['batched_voucher.json', 'export_configuration.json', 'fund_distribution.json', 'document.json', 'invoice_line.json', 'invoice.json', 'document_collection.json', 'document_data.json', 'document_metadata.json', 'adjustment.json', 'voucher_line_collection.json', 'batch_voucher_export.json', 'batch_group_collection.json', 'credentials.json', 'voucher_collection.json', 'voucher_line.json', 'invoice_collection.json', 'batched_voucher_line.json', 'invoice_line_collection.json', 'batch_group.json', 'voucher.json', 'batch_voucher.json']
DEBUG: lint-raml: configured_raml_files: ['invoice.raml', 'invoice-line-number.raml', 'invoice-number.raml', 'voucher.raml', 'voucher-number.raml']
DEBUG: lint-raml: found_raml_files: ['invoice.raml', 'voucher-number.raml', 'invoice-number.raml', 'voucher.raml', 'invoice-line-number.raml']
DEBUG: lint-raml: raml_files: ['invoice.raml', 'invoice-line-number.raml', 'invoice-number.raml', 'voucher.raml', 'voucher-number.raml']
INFO: lint-raml-schema: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-schema: Found 22 JSON schema files under directory './ramls/acq-models/mod-invoice-storage/schemas'
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: adjustment.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: batch_group.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: batch_group_collection.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: batch_voucher.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: batch_voucher_export.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: batched_voucher.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: batched_voucher_line.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: credentials.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: document.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: document_collection.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: document_data.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: document_metadata.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: export_configuration.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: fund_distribution.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: invoice.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: invoice_collection.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: invoice_line.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: invoice_line_collection.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: voucher.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: voucher_collection.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: voucher_line.json: Each property "description" is present.
DEBUG: lint-raml-schema: Doing jq
DEBUG: lint-raml-schema: voucher_line_collection.json: Each property "description" is present.
INFO: lint-raml: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
INFO: lint-raml: Did not detect any errors.

File: src/main/resources/templates/db_scripts/data-migration/invoices_table.sql [hunk 1]
```diff
@@ -1,3 +1,9 @@
 CREATE SEQUENCE IF NOT EXISTS ${myuniversity}_${mymodule}.invoice_number NO MAXVALUE START WITH 10000 CACHE 1 NO CYCLE;
 GRANT USAGE ON SEQUENCE ${myuniversity}_${mymodule}.invoice_number TO ${myuniversity}_${mymodule};
 CREATE UNIQUE INDEX IF NOT EXISTS invoices_invoice_number_unique_idx ON ${myuniversity}_${mymodule}.invoices ((jsonb->>'invoiceNumber'));
+
+UPDATE ${myuniversity}_${mymodule}.invoices
```
Comment (pautau-s) [inline src/main/resources/templates/db_scripts/data-migration/invoices_table.sql]: Now script section supports snippetPath, so as an improvement, you can put migration scripts into a separate file.
https://github.com/folio-org/mod-finance-storage/blob/49bf1480749e40fdb516d3f0a0057f1c8a1a3197/src/main/resources/templates/db_scripts/schema.json#L2-L8

Comment (craigmcnally) [reply src/main/resources/templates/db_scripts/data-migration/invoices_table.sql]: ah, cool... I thought it was crappy that we needed to do it this way so I'm glad there's a better way!

**Comment by javalkar-varun** [comment]
https://github.com/folio-org/mod-invoice-storage/blob/84bad103f2f0ec298bc6556d7caea756ff326ead/src/main/resources/data/vouchers/test_voucher.json#L5 needs to be removed as well since it no longer required

**Pull request merged**